src/ScottPlot/Plot/Plot.Axis.cs

```diff
@@ -352,7 +352,7 @@ public AxisLimits GetDataLimits(int xAxisIndex = 0, int yAxisIndex = 0)
 
             foreach (var plottable in GetPlottables())
             {
-                if (plottable.IsVisible == false)
+                if (plottable.IsVisible == false || plottable is Plottable.AxisLine || plottable is Plottable.AxisSpan)
```

Comment:
I think the issue here is that `AxisLine` and `AxisSpan` can (but don't always) have NaN limits. I think a NaN check would be a better solution, as it wouldn't exclude lines/spans with non-NaN limits.

If you're curious why NaN is such a problem, IEEE 754 requires NaN to evaluate as less than and greater than any number (including NaN). So if present it will be both the min and the max of any set.

Reply:
> I think a NaN check would be a better solution ... It would also catch any other plottables that could have NaN limits in the future.

Excellent suggestion, and I agree!

@EFeru if you modify this pull request I'd be happy to merge it in. You can revert the change you made so far, then these four assignments can be wrapped in if statements that check if the values they're comparing against are `!double.IsNaN()` - let me know if you hit any snags! Thanks again for this PR! 🚀 

https://github.com/ScottPlot/ScottPlot/blob/50627d532140a4dbed9a2db159667a0d171cfe33/src/ScottPlot/Plot/Plot.Axis.cs#L365-L375

Reply:
I did the suggested change. Tested and it works. Please have a look this was your intention of wrapping the check. I am not sure if you prefere an overall wrapping or individual wrapping for each xMin, xMax, etc.